### PR TITLE
Add `from_profiles_yml` to dbt `TargetConfigs`

### PR DESCRIPTION
**Pull request by kevingrismore**
This is a starting point for converting dbt's `profiles.yml` contents into Prefect blocks programmatically. Includes a few utilities for getting the profile contents from an expected default or user-supplied directory.

Also fixes cloud tests for `httpx==0.28.0` compatibility.

# Example usage
`profiles.yml`
```yaml
test:
  outputs:
    dev:
      type: duckdb
      path: dev.duckdb
      threads: 1

    prod:
      type: duckdb
      path: prod.duckdb
      threads: 4

  target: prod

test2:
  outputs:
    dev:
      type: duckdb
      path: dev.duckdb2
      threads: 1

    prod:
      type: duckdb
      path: prod.duckdb2
      threads: 4
    
  target: dev

test3:
  outputs:
    dev:
      dataset: demo
      job_execution_timeout_seconds: 300
      job_retries: 1
      keyfile: /path/to/keyfile.json
      location: US
      method: service-account
      priority: interactive
      project: sample_project
      threads: 1
      type: bigquery

    prod:
      dataset: prod
      job_execution_timeout_seconds: 300
      job_retries: 1
      keyfile: /path/to/keyfile.json
      location: US
      method: service-account
      priority: interactive
      project: sample_project
      threads: 1
      type: bigquery

  target: prod
```
```python
>>> from prefect_dbt.cli.configs import TargetConfigs
>>> TargetConfigs.from_profiles_yml() # first profile, default target
TargetConfigs(extras=None, allow_field_overrides=False, type='duckdb', threads=4)
>>> TargetConfigs.from_profiles_yml("test2") # selected profile, default target
TargetConfigs(extras=None, allow_field_overrides=False, type='duckdb', threads=1)
>>> TargetConfigs.from_profiles_yml("test3", "dev") # selected profile, selected target
TargetConfigs(extras={'job_execution_timeout_seconds': 300, 'job_retries': 1, 'keyfile': '/path/to/keyfile.json', 'location': 'US', 'method': 'service-account', 'priority': 'interactive', 'project': 'sample_project'}, allow_field_overrides=False, type='bigquery', threads=1)
>>> TargetConfigs.from_profiles_yml("test3", "dev").get_configs()
{'job_execution_timeout_seconds': 300, 'job_retries': 1, 'keyfile': '/path/to/keyfile.json', 'location': 'US', 'method': 'service-account', 'priority': 'interactive', 'project': 'sample_project', 'type': 'bigquery', 'schema': 'demo', 'threads': 1}
```

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py [hunk 1]
```diff
@@ -147,6 +148,79 @@ class TargetConfigs(BaseTargetConfigs):
...
+        target_name: Optional[str] = None,
+        profiles_dir: Optional[str] = None,
+        allow_field_overrides: bool = False,
+    ) -> "TargetConfigs":
+        """
+        Create a TargetConfigs instance from a dbt profiles.yml file.
+
+        Args:
+            profile_name: Name of the profile to use from profiles.yml.
+                         If None, uses the first profile.
```
Comment (zzstoatzz) [inline src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py:164]: format here looks a little goofy, if everyelse here is like that, then disregard this

**Review by zzstoatzz — APPROVED** [review]
one question about formatting but otherwise lgtm!

File: src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py [hunk 2]
```diff
@@ -147,6 +148,79 @@ class TargetConfigs(BaseTargetConfigs):
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/5zE9lxfzBHjw3tnEup4wWL/9a001902ed43a84c6c96d23b24622e19/dbt-bit_tm.png?h=250"  # noqa
     _documentation_url = "https://docs.prefect.io/integrations/prefect-dbt"  # noqa
 
+    @classmethod
+    def from_profiles_yml(
+        cls,
```
Comment (zzstoatzz) [inline src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py:153]: nit, `from typing_extensions import Self`
```suggestion
        cls: Self,
```

Comment (desertaxle) [reply src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py:153]: Since `from_profiles_yml` is a class method, would it be `Type[Self]`?

Comment (zzstoatzz) [reply src/integrations/prefect-dbt/prefect_dbt/cli/configs/base.py:153]: yes!

File: src/integrations/prefect-dbt/tests/test_utilities.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    assert get_profiles_dir() == expected
+
+
+def test_get_profiles_dir_from_env():
+    test_path = "/custom/path"
+    os.environ["DBT_PROFILES_DIR"] = test_path
+    try:
+        assert get_profiles_dir() == test_path
+    finally:
+        del os.environ["DBT_PROFILES_DIR"]
```
Comment (desertaxle) [inline src/integrations/prefect-dbt/tests/test_utilities.py:48]: You can use `monkeypatch.setenv` to simplify this a smidge:
```suggestion
def test_get_profiles_dir_from_env(monkeypatch):
    test_path = "/custom/path"
    monkeypatch.setenv("DBT_PROFILES_DIR", test_path)
    assert get_profiles_dir() == test_path
```

**Review by desertaxle — APPROVED** [review]
Nice!

**Pull request merged**